### ISSUE TITLE
Implement occlusion culling to reduce number of loaded chunks

### ISSUE THREAD

**Issue by cuchaz** [TallWorlds/CubicChunks#43]
yup

**Comment by kvverti** [comment]
Would this mean that a chunk must be visible for it to receive updates? Or would this only affect chunks sent to the player?

**Comment by cuchaz** [comment]
No. It will still get updated (on the server) as long as a player is nearby even if the player can't see it.

**Comment by mikesmiffy128** [comment]
Ah so this is just client side loading?

**Comment by cuchaz** [comment]
Yup

**Comment by kvverti** [comment]
This would mean that any entity tracking code remaining on the client would need to be moved to the server, as mobs would no longer be seen by the client outside the visible range. It would wreak havoc on the hostile mob cap.

**Comment by cuchaz** [comment]
It's going to be a long time before we start working on this feature. I wouldn't worry about those issues for now.

**Comment by mikesmiffy128** [comment]
Entity tracking should happen on the server anyway. It would avoid all the sync issues like with boats and what not.

**Comment by bluebear94** [comment]
As shown in [this blog post](https://tomcc.github.io/2014/08/31/visibility-1.html)?
